nixos/modules/misc/version.nix

```diff
@@ -82,8 +82,6 @@ in
 
     stateVersion = mkOption {
       type = types.str;
-      default = cfg.release;
```

Comment:
Was going to suggest `throw`ing a custom error in `default`, but that will obfuscate _which_ system is missing this info. The module system's built-in message includes the entire attribute path.
(This is assuming NixOS gets hooked into a deployment tool's module system correctly; can't vouch for each of them, but NixOps has an open PR that will accomplish this.)